apps/clientside-bot/test/utils.ts

```diff
@@ -0,0 +1,237 @@
+	return Array.from({ length: count as number }, () =>
+		typeof params === "function" ? creator(...params()) : creator(...params)
+	)
+}
+
+// this is to make sure that the IDs are unique
+const existingFAGCIds = new Set<string>()
+export const createFAGCId = (): string => {
+	const id = faker.random.alphaNumeric(6)
+	if (existingFAGCIds.has(id)) return createFAGCId()
+	existingFAGCIds.add(id)
+	return id
+}
+
+/**
+ * Get a random element from the provided array
+ */
+export const randomElementFromArray = <T>(array: T[]): T => {
+	return array[Math.floor(Math.random() * array.length)]
+}
+
+export const randomElementsFromArray = <T>(array: T[], count?: number): T[] => {
+	if (!count) {
+		count = Math.floor(Math.random() * array.length) + 1
+	}
+	return (
+		array
+			// sort the array randomly
+			.sort(() => 0.5 - Math.random())
+			// get the first count elements of the randomly sorted array
+			.slice(0, count)
+	)
+}
+
+const createDiscordId = (): string => {
+	return faker.datatype
+		.number({
+			min: 1e17, // discord user id
+			max: 1e19,
+		})
+		.toString()
+}
+
+export const createFAGCReport = ({
+	categoryIds,
+	communityIds,
+	playernames,
+	adminIds,
+}: {
+	categoryIds: string[]
+	communityIds: string[]
+	playernames?: string[]
+	adminIds?: string[]
+}): Report => {
+	return {
+		id: createFAGCId(),
+		playername: playernames
+			? randomElementFromArray(playernames)
+			: faker.internet.userName(),
+		categoryId: randomElementFromArray(categoryIds),
+		communityId: randomElementFromArray(communityIds),
+		description: faker.lorem.sentence(),
+		proof: faker.lorem.sentence(),
+		reportCreatedAt: faker.date.past(),
+		reportedTime: faker.date.past(),
+		automated: Math.random() > 0.5,
+		adminId: adminIds
+			? randomElementFromArray(adminIds)
+			: createDiscordId(),
+	}
+}
+
+export const createFAGCRevocation = ({
+	report,
+}: {
+	report: Report
+}): Revocation => {
+	return {
+		...report,
+		revokedAt: faker.date.past(),
+		revokedBy: createDiscordId(),
+	}
+}
+
+export const createGuildConfig = ({
+	categoryIds,
+	communityIds,
+	includeAllFilters,
+}: {
+	categoryIds: string[]
+	communityIds: string[]
+	includeAllFilters?: boolean
+}): GuildConfig => {
+	return {
+		guildId: createDiscordId(),
+		categoryFilters: includeAllFilters
+			? categoryIds
+			: randomElementsFromArray(
+					categoryIds,
+					Math.round(categoryIds.length / 2)
+			  ),
+		trustedCommunities: includeAllFilters
+			? categoryIds
+			: randomElementsFromArray(
+					communityIds,
+					Math.round(communityIds.length / 2)
+			  ),
+		roles: {
+			reports: createDiscordId(),
+			webhooks: createDiscordId(),
+			setConfig: createDiscordId(),
+			setCategories: createDiscordId(),
+			setCommunities: createDiscordId(),
+		},
+	}
+}
+
+export const createFAGCCategory = (): Category => {
+	return {
+		id: createFAGCId(),
+		name: faker.lorem.words(),
+		description: faker.lorem.sentence(),
+	}
```

Comment:
Do you correctly account for the maximum expected length of these datatypes when using `faker.lorem` ?